plugins/outputs/sql/sql.go

```diff
@@ -278,6 +351,15 @@ func (p *SQL) Write(metrics []telegraf.Metric) error {
 			values = append(values, value)
 		}
 
+		// modifying the table schema is opt-in
+		if p.TableUpdateTemplate != "" {
+			for i := range len(columns) {
+				if err := p.createColumn(tablename, columns[i], p.deriveDatatype(values[i])); err != nil {
+					return err
+				}
+			}
```

Comment:
Would it make sense to create all columns in one statement to reduce the number of queries...

Reply:
I had an earlier version doing this, but there was a major issue:

Unlike CREATE TABLE, you specify the options as an ADD COLUMN directive and repeat for every column

This complicates the string template, as parts of the query have to be repeated. And the user is required to enter a sort-of SQL query, which might end up more complex and confusing.

As far the number of queries goes... it's a hard one to gauge.

- Most plugins do not create new fields/tags
- ...even if they do, it's just a few new fields per flush

To take advantage and ensure there is >1 new field in a single flush:

- the user must use the merge aggregator
- as well as buffer more metrics, so there are more fields to process

Which is why i wasn't sure if it's worth it and ended up removing it.

Reply:
Ok, let's leave this optimization for later then!